prisma/seed.ts

```diff
@@ -51,12 +54,17 @@ async function main() {
     author: katerina,
     curationCategory: curationCategory1,
   });
+
+  // CollectionLabel table input data
+  const collectionLabelInputData: CreateCollectionLabelInput = {
+    collectionId: katerinaCollection.id,
```

Comment:
i am a bit confused why the `id` field is used to track the `label` and `collection` instead of the `externalId`?

Reply:
this also brings up the question why have both `id` and `externalId` and not keep just one?

Reply:
Good question @katerinachinnappan! We use this approach because we don't want the database id i.e `id` to be exposed to the clients/graphql. So we provide an `externalId` field for them to able to fetch specific item via `externalId`.

The internal `id` is then only used for joins/foreign keys. Since this table is merely a mapping table and used to join the other two tables, we remove the `externalId`.

Hope that makes sense!

Reply:
good questions!

it's a bad security practice to let auto-incrementing ids out into the world. potentially an attacker could enumerate your database this way. the `externalId` property is our mitigation strategy here (which is why it's used everywhere _except_ internal entity relationships).

at the design of this system, we decided to use auto-incrementing keys for internal relationships. it's more human-readable should someone need to manually inspect relationships, and the cross-table storage is slightly reduced. we didn't hold this position super strongly when designing, but decided to go with it. these kinds of decisions are always up for debate though.

tl;dr - `externalId` could be our one and only identifying field if we felt strongly.